doc/deployment-schema-1.0.json

```diff
@@ -0,0 +1,313 @@
+                                        "^.+$": {
+                                            "type": "string",
+                                            "examples": [
+                                                "FROM /* INTO $upstream"
+                                            ],
+                                            "pattern": "^(?i)FROM\\s\/.*?(\\s|\\sWHERE\\s.*?\\s)INTO\\s(\\$upstream|BrokeredEndpoint\\(.*?\\))$"
+                                        }
+                                    }
+                                },
+                                "storeAndForwardConfiguration": {
+                                    "type": "object",
+                                    "required": [
+                                        "timeToLiveSecs"
+                                    ],
+                                    "properties": {
+                                        "timeToLiveSecs": {
+                                            "type": "integer",
+                                            "examples": [
+                                                7200
+                                            ]
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            },
+            "patternProperties": {
+                "^[a-zA-Z][a-zA-Z0-9_.-]*$": {
+                    "type": "object",
+                    "required": [
+                        "properties.desired"
+                    ],
+                    "properties": {
+                        "properties.desired": {
+                            "type": "object"
+                        }
+                    }
+                }
+            },
+            "additionalProperties": false
+        }
+    },
+    "definitions": {
+        "moduleType": {
+            "enum": [
+                "docker"
+            ]
+        },
+        "status": {
+            "enum": [
+                "running",
+                "stopped"
+            ]
+        },
+        "restartPolicy": {
+            "enum": [
+                "never",
+                "on-failure",
+                "on-unhealthy",
+                "always"
+            ]
+        },
+        "moduleSettings": {
+            "type": "object",
+            "required": [
+                "image"
+            ],
+            "properties": {
+                "image": {
+                    "type": "string",
+                    "examples": [
+                        "mcr.microsoft.com/azureiotedge-agent:1.0"
+                    ]
+                },
+                "createOptions": {
+                    "$ref": "#/definitions/createOptions"
+                }
+            }
+        },
+        "env": {
+            "type": "object",
+            "patternProperties": {
+                "^[^\\+#$\\s]+$": {
```

Comment:
should be ^[^\\+#$\s]+$ ? It seems "s" could not be in the env name with the pattern.